lib/nm/index.ts

```diff
@@ -0,0 +1,121 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import NetworkManagerTypes from './types';
+
+const dbus: any = require('dbus-native');
+
+const systemBus: any = dbus.systemBus();
+const service: string[] = [ 'org', 'freedesktop', 'NetworkManager' ];
+
+export interface WifiDevice {
+	devicePath: string;
+	type: number;
+}
+
+export interface AccessPoint {
+	apPath: string;
+	apSsid: string;
+}
+
+export class NetworkManager extends NetworkManagerTypes {
+	wifiDevice: WifiDevice;
+	accessPoints: AccessPoint[];
+	wifiActive: boolean;
+
+	constructor() {
+		super();
+
+		this.wifiActive = true;
+
+		this.getWifiDevice()
+		.then((device: WifiDevice) => {
+			const service = Promise.promisifyAll(systemBus.getService('org.freedesktop.NetworkManager'));
+			service.getInterfaceAsync(device.devicePath, 'org.freedesktop.NetworkManager.Device.Wireless')
+			.then((iface) => {
+				console.log('YOLO');
```

Comment:
console.log

Reply:
Don't mind the console.log comments, I haven't seen this was a [WIP] pull request.

Reply:
Eheh, don't worry, I removed them anyway.
Also "YOLO" isn't clearly the best log to see, as well as basically all the other logs I usually put when developing :P